### PR TITLE
{Role} `az role assignment create`: Show warning if `--scope` argument is not specified

### PR DESCRIPTION
**Pull request by jiasli**
**Description**
Similar to #20965
A temporary warning for #24753

For `az role assignment create`, `--scope` defaults to the subscription. Without explicit consent from the user on the `--scope`, this behavior is considered Elevation of Privilege.

This PR adds a warning if `--scope` is not specified:

```
> az role assignment create --assignee 5963f50c-7c43-405c-af7e-53294de76abd --role Reader --resource-group clitest.rgizxewh7npwdc63xpadmmi73hvq5ftcza2acf5wiyulxnjy5fgujpvvqll63kndkru
Starting from Azure CLI 2.47.0, --scopes argument will become required for creating role assignments. Please explicitly specify --scopes.
{
  "canDelegate": null,
  "condition": null,
  "conditionVersion": null,
  "description": null,
  "id": "/subscriptions/0b1f6471-1bf0-4dda-aec3-cb9272f09590/resourceGroups/clitest.rgizxewh7npwdc63xpadmmi73hvq5ftcza2acf5wiyulxnjy5fgujpvvqll63kndkru/providers/Microsoft.Authorization/roleAssignments/3a36df35-93c4-4e5b-b605-61e9f9a0f1fc",
  "name": "3a36df35-93c4-4e5b-b605-61e9f9a0f1fc",
  "principalId": "5963f50c-7c43-405c-af7e-53294de76abd",
  "principalName": "admin2@AzureSDKTeam.onmicrosoft.com",
  "principalType": "User",
  "resourceGroup": "clitest.rgizxewh7npwdc63xpadmmi73hvq5ftcza2acf5wiyulxnjy5fgujpvvqll63kndkru",
  "roleDefinitionId": "/subscriptions/0b1f6471-1bf0-4dda-aec3-cb9272f09590/providers/Microsoft.Authorization/roleDefinitions/acdd72a7-3385-48ef-bd42-f606fba81ae7",
  "roleDefinitionName": "Reader",
  "scope": "/subscriptions/0b1f6471-1bf0-4dda-aec3-cb9272f09590/resourceGroups/clitest.rgizxewh7npwdc63xpadmmi73hvq5ftcza2acf5wiyulxnjy5fgujpvvqll63kndkru",
  "type": "Microsoft.Authorization/roleAssignments"
}
```

Even if `--resource-group` or `--subscription` is specified, we still show this warning because as described in #24753, we may drop `--resource-group` and `--subscription`, so `--scope` should always be used as the unified, all-in-one argument for specifying the role assignment scope.

**History Notes**

[Role] `az role assignment create`: Show warning if `--scope` argument is not specified: `--scope` argument will become required for creating a role assignment in the breaking change release of the fall of 2023. Please explicitly specify `--scope`.

### CODE REVIEW

File: src/azure-cli/azure/cli/command_modules/role/custom.py [hunk 1]
```diff
@@ -50,6 +50,9 @@
     "The output includes credentials that you must protect. Be sure that you do not include these credentials in "
     "your code or check the credentials into your source control. For more information, see https://aka.ms/azadsp-cli")
 
+SCOPE_WARNING = "In a future release, --scope argument will become required for creating a role assignment. " \
```
Comment (jiasli) [inline src/azure-cli/azure/cli/command_modules/role/custom.py:53]: Similar to #20965, we can't give an exact version yet.

File: src/azure-cli/azure/cli/command_modules/role/_help.py [hunk 2]
```diff
@@ -685,22 +685,21 @@
 helps['role assignment create'] = """
 type: command
 short-summary: Create a new role assignment for a user, group, or service principal.
+long-summary: In a future release, --scope argument will become required for creating a role assignment. Please explicitly specify --scope.
 examples:
-  - name: Create role assignment for an assignee.
-    text: az role assignment create --assignee sp_name --role a_role
+  - name: Create role assignment to grant an assignee Reader role on a virtual machine.
```
Comment (jiasli) [inline src/azure-cli/azure/cli/command_modules/role/_help.py:690]: I find this description cumbersome. @dcaro, @dbradish-microsoft, any though on how we can improve it?

File: src/azure-cli/azure/cli/command_modules/role/_help.py [hunk 3]
```diff
@@ -685,22 +685,21 @@
 helps['role assignment create'] = """
 type: command
 short-summary: Create a new role assignment for a user, group, or service principal.
+long-summary: In a future release, --scope argument will become required for creating a role assignment. Please explicitly specify --scope.
```
Comment (dcaro) [inline src/azure-cli/azure/cli/command_modules/role/_help.py:688]: @jiasli We should be specific about the timeline.

File: src/azure-cli/azure/cli/command_modules/role/_help.py (hunk 2 above)
Comment (dcaro) [reply src/azure-cli/azure/cli/command_modules/role/_help.py:690]:
```suggestion
  - name: Create role assignment to grant Reader role to the specified assignee on an Azure virtual machine
```
I am not sure if we have any guidelines for the length of this field so I tried to keep is short. We can be more explicit if we have more space.

**Review by dcaro — COMMENTED** [review]
@jiasli just a suggestion about the name. 
We need to give a more specific timeline.

File: src/azure-cli/azure/cli/command_modules/role/_help.py (hunk 3 above)
Comment (dcaro) [reply src/azure-cli/azure/cli/command_modules/role/_help.py:688]:
```suggestion
long-summary: --scope argument will become required for creating a role assignment in the breaking change release of the fall of 2023. Please explicitly specify --scope.
```

File: src/azure-cli/azure/cli/command_modules/role/_help.py (hunk 2 above)
Comment (dbradish-microsoft) [reply src/azure-cli/azure/cli/command_modules/role/_help.py:690]:
```suggestion
  - name: Create role assignment to grant the specified assignee the Reader role on an Azure virtual machine.
```

File: src/azure-cli/azure/cli/command_modules/role/_help.py [hunk 4]
```diff
@@ -685,22 +685,21 @@
...
     text: >-
-        az role assignment create --role "Owner" --assignee "John.Doe@Contoso.com"
+        az role assignment create --role Owner
+        --scope /subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/MyResourceGroup/providers/Microsoft.Storage/storageAccounts/MyStorageAccount
+        --assignee "John.Doe@Contoso.com"
         --description "Role assignment foo to check on bar"
         --condition "@Resource[Microsoft.Storage/storageAccounts/blobServices/containers:Name] stringEquals 'foo'"
         --condition-version "2.0"
     supported-profiles: latest
-  - name: Create a new role assignment for a user, group, or service principal. (autogenerated)
```
Comment (dbradish-microsoft) [inline src/azure-cli/azure/cli/command_modules/role/_help.py:698]: @jiasli , in removing this autogenerated line, won't it just come back in again in the next Aladdin PR ([example](https://github.com/Azure/azure-cli/pull/25137)) that backfills the short and long descriptions, and examples in _help.py?

File: src/azure-cli/azure/cli/command_modules/role/custom.py (hunk 1 above)
Comment (dbradish-microsoft) [reply src/azure-cli/azure/cli/command_modules/role/custom.py:53]: @jiasli , but can't we at least give the anticipated breaking change (BK) window as are BKs are moving to a bi-annual schedule?

**Review by dbradish-microsoft — CHANGES_REQUESTED** [review]

File: src/azure-cli/azure/cli/command_modules/role/custom.py (hunk 1 above)
Comment (dcaro) [reply src/azure-cli/azure/cli/command_modules/role/custom.py:53]: this should also be consistent with the content on line 688.
We can mention something like "breaking change of Fall of 2023".

File: src/azure-cli/azure/cli/command_modules/role/_help.py (hunk 4 above)
Comment (jiasli) [reply src/azure-cli/azure/cli/command_modules/role/_help.py:698]: Based on my understanding, Aladdin-generated content is merged into `_help.py` on-demand and won't be automatically done.

Comment (dbradish-microsoft) [reply src/azure-cli/azure/cli/command_modules/role/_help.py:698]: @jiasli, "on demand" = someone in engineering is reviewing each Aladdin PR change and accepting/manually implementing(?) it?  I have a lot of questions on the Aladdin process, and this really isn't the place so as long as your updates won't be over-written, we can move on.

**Comment by jiasli** [comment]
We will also need to refine the conceptual doc: https://learn.microsoft.com/en-us/azure/role-based-access-control/role-assignments-cli

File: src/azure-cli/azure/cli/command_modules/role/_help.py (hunk 4 above)
Comment (jiasli) [reply src/azure-cli/azure/cli/command_modules/role/_help.py:698]: Correct. Someone needs to manually accept it.

**Review by dbradish-microsoft — APPROVED** [review]

**Review by yonzhan — APPROVED** [review]

**Review by evelyn-ys — APPROVED** [review]

**Review by dcaro — APPROVED** [review]
LGTM

**Pull request merged**